### PR TITLE
Added a check for empty array in _shared.utils.py

### PR DESCRIPTION
**Pull request by tseclaudia**
## Description
Addresses Issue #2053

### CODE REVIEW

**Comment by soupault** [comment]
@tseclaudia Hi! Thank for catching this up!
In order to close #2053, you'll need to address the 2nd part of the issue as well: `adding assert_nD to feature.local_binary_pattern`. Should also add a simple test which fails on a current `master` but will be fixed with your patch.

**Comment by tseclaudia** [comment]
@soupault Doesn't feature.local_binary_pattern already contain [assert_nD](https://github.com/scikit-image/scikit-image/blob/master/skimage/feature/texture.py#L310) which checks that the passed in array meets the desired ndims and isn't an empty array? Does that correctly address the second part of the issue?

**Comment by soupault** [comment]
@tseclaudia fair enough, sorry. OK, then I'd like to ask you to add a test to `_shared/tests/test_utils.py` checking if `assert_nD` raises an error on empty array.
Oh, and, please, follow code style conventions (i.e. wrap at 80th column).

File: skimage/_shared/tests/test_utils.py [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
 import numpy.testing as npt
+import unittest
 
 
+
+def test_assert_nD():
+    try:
+        z = np.random.random(200**2).reshape((200, 200))
+        x = z[10:30, 30:10]
+        assert_nD(x, 2)
```
Comment (soupault) [inline skimage/_shared/tests/test_utils.py]: There is no need to use `try/except` here. Just use `numpy.testing.assert_raises`, i.e.
```
from numpy.testing import assert_raises
...
assert_raises(ValueError, assert_nD, x, 2)
```

File: skimage/feature/tests/test_texture.py [hunk 2]
```diff
@@ -228,6 +229,22 @@ def test_uniform(self):
                         [2, 9, 0, 8, 1, 2]])
         np.testing.assert_array_equal(lbp, ref)
 
+    def test_empty(self):
+        # with unittest.TestCase.assertRaises(ValueError):
+        #     z = np.random.random(200**2).reshape((200, 200))
+        #     x = z[10:30, 30:10]
+        #     lbp = feature.local_binary_pattern(x, 8, 1, 'uniform')
+        try:
```
Comment (soupault) [inline skimage/feature/tests/test_texture.py]: Same here.

File: skimage/feature/tests/test_texture.py [hunk 3]
```diff
@@ -228,6 +229,22 @@ def test_uniform(self):
                         [2, 9, 0, 8, 1, 2]])
         np.testing.assert_array_equal(lbp, ref)
 
+    def test_empty(self):
+        # with unittest.TestCase.assertRaises(ValueError):
```
Comment (soupault) [inline skimage/feature/tests/test_texture.py]: Remove the comments.

File: skimage/_shared/utils.py [hunk 4]
```diff
@@ -147,7 +147,7 @@ def safe_as_int(val, atol=1e-3):
 
 def assert_nD(array, ndim, arg_name='image'):
     """
-    Verify an array meets the desired ndims.
+    Verify an array meets the desired ndims and array isn't empty.
```
Comment (soupault) [inline skimage/_shared/utils.py]: Did you accidentally remove the proposed fix?

File: skimage/_shared/tests/test_utils.py [hunk 5]
```diff
@@ -1,7 +1,17 @@
...
+from skimage._shared.utils import (copy_func, 
+                                    assert_nD)
 import numpy.testing as npt
+from numpy.testing import assert_raises 
 
 
+
+
+def test_assert_nD():
+    try:
```
Comment (soupault) [inline skimage/_shared/tests/test_utils.py]: Remove `try:`

File: skimage/_shared/tests/test_utils.py [hunk 6]
```diff
@@ -1,7 +1,17 @@
...
+        z = np.random.random(200**2).reshape((200, 200))
+        x = z[10:30, 30:10]
+        assert_raises(ValueError, assert_nDx, 2)
```
Comment (soupault) [inline skimage/_shared/tests/test_utils.py]: -> `assert_raises(ValueError, assert_nD, x, 2)`

File: skimage/_shared/tests/test_utils.py [hunk 7]
```diff
@@ -1,7 +1,16 @@
-from skimage._shared.utils import copy_func
+from skimage._shared.utils import (copy_func,
```
Comment (jni) [inline skimage/_shared/tests/test_utils.py]: one line: `from skimage._shared.utils import copy_func, assert_nD`

File: skimage/_shared/tests/test_utils.py [hunk 8]
```diff
@@ -1,7 +1,16 @@
...
 import numpy.testing as npt
+from numpy.testing import assert_raises 
 
 
+
+
+def test_assert_nD():
+    z = np.random.random(200**2).reshape((200, 200))
+    x = z[10:30, 30:10]
+    assert_raises(ValueError, assert_nD, 2)
```
Comment (jni) [inline skimage/_shared/tests/test_utils.py]: Should this be `assert_nD, x, 2`?

**Comment by soupault** [comment]
Thanks @tseclaudia!

**Pull request merged**